gpmp2/kinematics/tests/testWorkspacePoseVelocityPrior.cpp

```diff
@@ -0,0 +1,104 @@
+/**
+ *  @file    testWorkspacePoseVelocityPrior.cpp
+ *  @author  Matthew King-Smith
+ **/
+
+#include <CppUnitLite/TestHarness.h>
+#include <gpmp2/kinematics/WorkspacePoseVelocityPrior.h>
+#include <gtsam/base/Matrix.h>
+#include <gtsam/base/Testable.h>
+#include <gtsam/base/numericalDerivative.h>
+#include <gtsam/inference/Symbol.h>
+#include <gtsam/nonlinear/LevenbergMarquardtOptimizer.h>
+#include <gtsam/nonlinear/NonlinearFactorGraph.h>
+#include <gtsam/nonlinear/Values.h>
+
+#include <iostream>
+
+using namespace std;
+using namespace gtsam;
+using namespace gpmp2;
+
+/* ************************************************************************** */
+TEST(WorkspacePoseVelocityPrior, error) {
+  Vector2 a(1, 1), alpha(M_PI / 2, 0), d(0, 0);
+  Arm arm = Arm(2, a, alpha, d);
+  Vector2 q;
+  Vector2 qdot;
+  Pose3 des_pose;
+  // Vector6 des_vel;
+  // WorkspacePoseVelocityPrior factor;
+  Vector actual, expect;
+  Matrix H_pose_exp, H_pose_act, H_vel_exp, H_vel_act;
+  noiseModel::Gaussian::shared_ptr cost_model =
+      noiseModel::Isotropic::Sigma(12, 1.0);
+
+  q = Vector2(M_PI / 4.0, -M_PI / 2);
+  qdot = (Vector2() << 0.1, 0.1).finished();
+  des_pose = Pose3();
+  Vector6 des_vel = (Vector6() << 0, 0, 0, 0, 0, 0).finished();
+  WorkspacePoseVelocityPrior factor(0, 0, cost_model, arm, des_pose, des_vel);
+  actual = factor.evaluateError(q, qdot, &H_pose_act, &H_vel_act);
+  expect = (Vector(12) << 00.613943126, 1.48218982, -0.613943126, 1.1609828,
```

Comment:
Can you please comment where you got these numbers from? And please mark this as a regression test.